### PR TITLE
feat(cli): add 'hyperlane core read' command

### PR DESCRIPTION
**Pull request by nbayindirli**
### Description

* adds new `hyperlane core read` command
* this replaces previous `hl ism read` & `hl hook read` control flows

### Related issues

- fixes https://github.com/hyperlane-xyz/issues/issues/1185

### Backward compatibility

* yes

### Testing

* [x] manual
* [ ] ci-test

### CODE REVIEW

File: typescript/cli/src/commands/config.ts [hunk 1]
```diff
@@ -38,13 +31,7 @@ const createCommand: CommandModule = {
   command: 'create',
   describe: 'Create a new Hyperlane config',
   builder: (yargs) =>
-    yargs
-      .command(createChainConfigCommand)
-      .command(createIsmConfigCommand)
-      .command(createHookConfigCommand)
-      .command(createWarpRouteDeployConfigCommand)
-      .version(false)
-      .demandCommand(),
```
Comment (yorhodes) [inline typescript/cli/src/commands/config.ts:47]: did we decide we won't alias here?

File: typescript/cli/src/commands/core.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      ...chainCommandOption,
+      demandOption: true,
+    },
+    ismAddress: addressCommandOption(
+      'Address of the Interchain Security Module to read.',
+      false,
+    ),
+    hookAddress: addressCommandOption('Address of the Hook to read.', false),
+    ismOut: outputFileCommandOption(),
+    hookOut: outputFileCommandOption(),
```
Comment (yorhodes) [inline typescript/cli/src/commands/core.ts:78]: this should read the core config as one with just mailbox address

File: typescript/cli/src/commands/warp.ts
Comment (yorhodes) [inline typescript/cli/src/commands/warp.ts:1]: did you mean to include this here?

File: typescript/cli/src/config/ism.ts [hunk 3]
```diff
@@ -236,17 +250,22 @@ export async function createMultisigConfig(
...
 }
 
-async function createTrustedRelayerConfig(): Promise<ZodIsmConfig> {
-  const relayer = await input({
-    message: 'Enter relayer address',
-  });
+export async function createTrustedRelayerConfig(
+  context: CommandContext,
+): Promise<ZodIsmConfig> {
+  const relayer = await detectAndConfirmOrPrompt(
```
Comment (yorhodes) [inline typescript/cli/src/config/ism.ts:256]: love it

File: typescript/cli/src/config/warp.ts [hunk 4]
```diff
@@ -174,3 +195,23 @@ export function readWarpRouteConfig(filePath: string): WarpCoreConfig {
...
+) {
+  return {
+    type: IsmType.AGGREGATION,
+    modules: [
+      await createTrustedRelayerConfig(context),
+      await createRoutingConfig(
+        context,
+        IsmType.FALLBACK_ROUTING,
+        remote,
+        origins,
```
Comment (yorhodes) [inline typescript/cli/src/config/warp.ts:212]: I think we want origins to be empty here

File: typescript/cli/src/commands/core.ts (hunk 2 above)
Comment (yorhodes) [reply typescript/cli/src/commands/core.ts:78]: also can punt this until we change the core config command to build the same shape

**Review by yorhodes — APPROVED** [review]

File: typescript/cli/src/commands/core.ts (hunk 2 above)
Comment (nbayindirli) [reply typescript/cli/src/commands/core.ts:78]: created issue to capture: https://github.com/hyperlane-xyz/issues/issues/1282

File: typescript/cli/src/config/warp.ts (hunk 4 above)
Comment (nbayindirli) [reply typescript/cli/src/config/warp.ts]: this is empty on cli-2.0– i think rebase was out of date when you saw this

File: typescript/cli/src/commands/warp.ts
Comment (nbayindirli) [reply typescript/cli/src/commands/warp.ts:1]: rebase was just out of date

**Pull request merged**